src/Chapter1.hs

```diff
@@ -537,7 +551,12 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid :: Ord a => a -> a -> a -> a
+mid x y z 
+  | (y < x && x < z) || (z < x && x < y) || (x == y && x /= z) || (x == z && x /= y) = x
+  | (x < y && y < z) || (z < y && y < x) || (y == x && y /= z) || (y == z && y /= x) = y
+  | (x < z && z < y) || (y < z && z < x) || (z == x && z /= y) || (z == y && z /= x) = z
+  | x == y && y == z = x -- all the same
```

Comment:
As we mentioned, the compiler in Haskell is very attentive to the exhaustive pattern-matching. And here it would warn you that `Pattern matching is not exhaustive`, as the guards have quite complicated logic, and the compiler won't be able to prove that it covers all the cases.

Because of that, you will need to add another guard – `| otherwise = z`, to tell the compiler, that your pattern matching is exhaustive 🙂
But in these case, you can just combine the last two of your cases into the single `| otherwise = z` 🙂